service-base/src/main/java/org/eclipse/hono/service/command/CommandConnection.java

```diff
@@ -40,10 +40,20 @@
      *         the link cannot be established, e.g. because this client is not connected.
      * @throws NullPointerException if tenantId, deviceId or messageConsumer is {@code null}.
      */
-    Future<MessageConsumer> createCommandConsumer(String tenantId, String deviceId,
+    Future<MessageConsumer> getOrCreateCommandConsumer(String tenantId, String deviceId,
             BiConsumer<ProtonDelivery, Message> commandConsumer,
             Handler<Void> closeHandler);
 
+    /**
+     * Close the command receiver (and removes it from internal map).
+     *
+     * @param tenantId The tenant to consume commands from.
+     * @param deviceId The device for which the consumer will be created.
+     * @return A future indicating the result of the closing operation.
+     * @throws NullPointerException if tenantId or deviceId is {@code null}.
+     */
+    Future<Void> closeCommandReceiver(String tenantId, String deviceId);
```

Comment:
Either `consumer`, or `receiver`.
Currently these two words are used at the same time, while denoting the same, right?

Reply:
Since we have used `sender` below, `receiver` might be the better counterpart. WDYT?

Reply:
At other parts of Hono "consumer" is used and also the existing method was called with consumer.., so I would stick with it.

Reply:
Fine, but shouldn't then this method be renamed to `closeCommandConsumer` ?

Reply:
Sure - that is what I did. :)